### PR TITLE
Feature/df 1892 triage

### PR DESCRIPTION
**Pull request by ganqqwerty**

### CODE REVIEW

File: alv-portal-ui/src/app/shared/backend-services/competence-catalog/competence-element/competence-element.repository.ts [hunk 1]
```diff
@@ -2,47 +2,58 @@ import { Injectable } from '@angular/core';
...
-    return this.http.get<CompetenceElement>(this.resourceUrl + id);
+    return this.triageService.endpoint$.pipe(
+      switchMap(endpoint => this.http.get<CompetenceElement>(endpoint + this.resourceUrl + id))
+    );
+
   }
 
   findByIds(ids: string[]): Observable<CompetenceElement[]> {
-    return this.http.post<CompetenceElement[]>(`${this.findUrl}/byIds`, ids);
+    return this.http.post<CompetenceElement[]>(KK_EDITOR_ENDPOINT + `${this.findUrl}/byIds`, ids);
```
Comment (emina-str) [inline alv-portal-ui/src/app/shared/backend-services/competence-catalog/competence-element/competence-element.repository.ts]: endpoint should be defined through triageService

File: alv-portal-ui/src/app/shared/backend-services/competence-catalog/competence-element/competence-element.repository.ts [hunk 2]
```diff
@@ -2,47 +2,58 @@ import { Injectable } from '@angular/core';
...
   }
 
   search(request: PagedSearchRequest): Observable<Page<CompetenceElement>> {
     const params = createPageableURLSearchParams(request);
-    return this.http.post<Page<CompetenceElement>>(this.searchUrl, request.body, {
+    return this.http.post<Page<CompetenceElement>>(KK_EDITOR_ENDPOINT + this.searchUrl, request.body, {
```
Comment (emina-str) [inline alv-portal-ui/src/app/shared/backend-services/competence-catalog/competence-element/competence-element.repository.ts]: endpoint should be defined through triageService

File: alv-portal-ui/src/app/shared/backend-services/competence-catalog/competence-set/competence-set.repository.ts [hunk 3]
```diff
@@ -4,67 +4,79 @@ import { Observable } from 'rxjs';
...
+        endpoint => this.http.get<CompetenceSetSearchResult[]>(endpoint + `${this.findUrl}/byCompetenceElementId`, {
+          params: new HttpParams().set('id', competenceElementId)
+        })
+      ));
   }
 
   search(request: PagedSearchRequest): Observable<Page<CompetenceSetSearchResult>> {
     const params = createPageableURLSearchParams(request);
-    return this.http.post<Page<CompetenceSetSearchResult>>(this.searchUrl, request.body, {
+    return this.http.post<Page<CompetenceSetSearchResult>>(KK_EDITOR_ENDPOINT + this.searchUrl, request.body, {
```
Comment (emina-str) [inline alv-portal-ui/src/app/shared/backend-services/competence-catalog/competence-set/competence-set.repository.ts]: endpoint should be defined through triageService

File: alv-portal-ui/src/app/shared/backend-services/competence-catalog/triage.service.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class TriageService {
+
+  endpoint$: Observable<KkEndpoint>;
+
+  constructor(private authenticationService: AuthenticationService) {
+    this.endpoint$ = this.authenticationService.getCurrentUser().pipe(
+      map(user => !!user && (user.isCompetenceCatalogEditor() || user.isAdmin()) ? KK_EDITOR_ENDPOINT : KK_PUBLIC_ENDPOINT)
```
Comment (emina-str) [inline alv-portal-ui/src/app/shared/backend-services/competence-catalog/triage.service.ts]: @cjpsycos  Could you have a look at this? Is the part for `user.isAdmin()` OK?

Comment (cjpsycos) [reply alv-portal-ui/src/app/shared/backend-services/competence-catalog/triage.service.ts]: only editor and sysAdmin can land on the editor-endpoint

File: alv-portal-ui/src/app/shared/backend-services/competence-catalog/triage.service.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import { AuthenticationService } from '../../../core/auth/authentication.service';
+import { Observable } from 'rxjs';
+import { KK_EDITOR_ENDPOINT, KK_PUBLIC_ENDPOINT, KkEndpoint } from './endpoints';
+import { map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TriageService {
```
Comment (cjpsycos) [inline alv-portal-ui/src/app/shared/backend-services/competence-catalog/triage.service.ts]: triage is not a good name. it is more routing

**Pull request merged**